scripts/typescript_checks_test.py

```diff
@@ -112,12 +112,43 @@ def mock_popen_for_errors(
                 typescript_checks.compile_and_check_typescript(
                     typescript_checks.TSCONFIG_FILEPATH)
 
-    def test_no_error_for_invalid_compilation_of_strict_tsconfig(self) -> None:
+    def test_error_is_raised_for_invalid_compilation_of_strict_tsconfig(
+            self) -> None:
         """Test that error is produced if stdout is not empty."""
-        typescript_checks.TS_STRICT_EXCLUDE_PATHS = []
-        with self.assertRaisesRegex(SystemExit, '1'):
-            typescript_checks.compile_and_check_typescript(
-                typescript_checks.STRICT_TSCONFIG_FILEPATH)
+        with self.swap(typescript_checks, 'TS_STRICT_EXCLUDE_PATHS', []):
+            with self.assertRaisesRegex(SystemExit, '1'):
+                typescript_checks.compile_and_check_typescript(
+                    typescript_checks.STRICT_TSCONFIG_FILEPATH)
+
+    def test_error_is_raised_for_invalid_compilation_of_temp_strict_tsconfig(
+            self) -> None:
+        """Test that error is produced if stdout is not empty."""
+        class MockOuput:
+            def __init__(self, call_counter: int = 0) -> None:
+                self.call_counter = call_counter
+            def readline(self) -> str: # pylint: disable=missing-docstring
+                self.call_counter = self.call_counter + 1
+                if self.call_counter == 1:
+                    return 'core/templates/App.ts'
+                elif self.call_counter == 2:
+                    return 'core/new_directory/new_file.ts'
+                else:
+                    return ''
```

Comment:
If this branch isn't expected, raise an exception.

Also, this is a good use of the call_counter (though I still don't think you need it in setUp, you can define it as a local variable in this test, right)?

Reply:
No this is intentional since `iter()` method defined in script [[ref](https://github.com/oppia/oppia/blob/develop/scripts/typescript_checks.py#L608)] calls the `readline()` method until it returns `''`.

Reply:
Then please explain it. You can do that in the docstring: "This class simulates a stdout that produces the following output: A, B, C".

Also for mocks like this (here and elsewhere in your PR): if you expect them to only be called 3 times, do something like:

```
    return_values = {
        1: 'core/templates/App.ts',
        2: 'core/new_directory/new_file.ts',
        3: ''
    }
```

Then an exception will be thrown if an invalid call_counter is passed as the key, which indicates that something unexpected is happening with the test.

Reply:
Added docstrings.